cpp/src/reductions/scan/scan.cpp

```diff
@@ -23,33 +23,45 @@
 
 namespace cudf {
 
+namespace detail {
 std::unique_ptr<column> scan(column_view const& input,
                              std::unique_ptr<scan_aggregation> const& agg,
                              scan_type inclusive,
                              null_policy null_handling,
+                             rmm::cuda_stream_view stream,
```

Comment:
This API is a great example of why we need to isolate stream handling at the public/detail barrier. This function uses a stream value 5 times, and relied on `cudf::default_stream_value` because there was no detail API. I converted the public API to a detail API and use a single source for the stream, defined in the public API.